src/journal/JournalRecorder.h

```diff
@@ -22,6 +23,16 @@ namespace journal {
 
 class JournalRecorder {
 public:
+  struct ObjectRecorderHolder {
```

Comment:
I was thinking that perhaps you could just expose the ```ObjectRecorder```'s lock via a new get method and create a new ```ObjectRecorder::append_unlock(const AppendBuffers &append_buffers, Mutex &lock)``` method that ```JournalRecorder``` would invoke -- and ```ObjectRecorder``` would be responsible for unlocking once safe to do so.